python/pyspark/sql/functions.pyi

```diff
@@ -137,6 +137,8 @@ def sha1(col: ColumnOrName) -> Column: ...
 def sha2(col: ColumnOrName, numBits: int) -> Column: ...
 def hash(*cols: ColumnOrName) -> Column: ...
 def xxhash64(*cols: ColumnOrName) -> Column: ...
+def assert_true(col: ColumnOrName, errMsg: Union[Column, str] = ...): ...
```

Comment:
Two small notes (sorry for being late):

- I think we should annotate return type for `assert_true` - it will type check because of implicit `Any`, but I think it is better to avoid such cases

   ```python
   def assert_true(col: ColumnOrName, errMsg: Union[Column, str] = ...) -> Column: ...
   ```

- For `def raise_error` I'd use [`NoReturn`](https://docs.python.org/3/library/typing.html#typing.NoReturn):

    ```python
    from typing import NoReturn

    def raise_error(errMsg: Union[Column, str]) -> NoReturn: ...
    ```

Reply:
@karenfeng Could you fix them above in followup?

Reply:
> * For `def raise_error` I'd use [`NoReturn`](https://docs.python.org/3/library/typing.html#typing.NoReturn):

This might indicate intention here, though technically speaking it's still a `Column`, so

```python
def raise_error(errMsg: Union[Column, str]) -> Column: ...
```

is still correct (and literal one). Do you have any thoughts about it @HyukjinKwon?

Reply:
I think doing `Column` is fine.